rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -2078,31 +2102,17 @@ void game_list_frame::BatchRemoveCustomPadConfigurations()
 		return;
 	}
 
-	progress_dialog* pdlg = new progress_dialog(tr("Custom Pad Configuration Batch Removal"), tr("Removing all custom pad configurations"), tr("Cancel"), 0, total, true, this);
-	pdlg->setAutoClose(false);
-	pdlg->setAutoReset(false);
-	pdlg->show();
+	progress_dialog* pdlg = new progress_dialog(tr("Custom Pad Configuration Batch Removal"), tr("Removing all custom pad configurations"), tr("Cancel"), 0, total, false, this);
 
-	u32 removed = 0;
-	for (const auto& serial : serials)
-	{
-		if (pdlg->wasCanceled())
+	BatchActionBySerials(pdlg, serials, tr("%0/%1 custom pad configurations cleared"),
```

Comment:
I think this is all way too complex and detached from the reading flow.

Just create a (new QFutureWatcher(this)) and connect it's resultReady signal to the pdlg::setValue.
Then connect the finish signal to deleteLater of QFutureWatcher and stuff.

This would keep the code contained in this function without too much overhead instead of having one huge complex thing a couple of hundred lines above

Reply:
I added asymcronous waiting for emulation close for batch PPU compilation, not sure if it would still fit QFutureWatcher anymore.